workers/unity/Packages/com.improbable.gdk.core/Utility/ReflectionUtility.cs

```diff
@@ -0,0 +1,147 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    internal static class ReflectionUtility
+    {
+        private static List<Type> allNonAbstractTypes;
+
+        private static readonly Dictionary<InterfaceAndAttributes, List<Type>> propertiesToType =
```

Comment:
Why is this cached as well?

Reply:
This one is the cache you actually want. It means creating something like a send message object doesn't iterate over everything after startup.